lib/Search/BasicSearch/BasicSearchEngine.php

```diff
@@ -0,0 +1,526 @@
+    }
+
+    /**
+     * @return array
+     */
+    protected function getSearchModules(): array
+    {
+        $unifiedSearchModuleDisplay = $this->getUnifiedSearchModulesDisplay();
+
+        require_once 'include/ListView/ListViewSmarty.php';
+
+        global $beanList, $current_user;
+
+        if (!empty($_REQUEST['advanced']) && $_REQUEST['advanced'] !== 'false') {
+            $modulesToSearch = [];
+            if (!empty($_REQUEST['search_modules'])) {
+                foreach (explode(',', $_REQUEST['search_modules']) as $key) {
+                    if (isset($unifiedSearchModuleDisplay[$key]) && !empty($unifiedSearchModuleDisplay[$key]['visible'])) {
+                        $modulesToSearch[$key] = $beanList[$key];
+                    }
+                }
+            }
+
+            $current_user->setPreference('showGSDiv', $_REQUEST['showGSDiv'] ?? 'no', 0, 'search');
+            $current_user->setPreference('globalSearch', $modulesToSearch, 0, 'search');
+        } else {
+            $users_modules = $current_user->getPreference('globalSearch', 'search');
+            $modulesToSearch = [];
+
+            if (!empty($users_modules)) {
+                // Use user's previous selections
+                foreach ($users_modules as $key => $value) {
+                    if (isset($unifiedSearchModuleDisplay[$key]) && !empty($unifiedSearchModuleDisplay[$key]['visible'])) {
+                        $modulesToSearch[$key] = $beanList[$key];
+                    }
+                }
+            } else {
+                foreach ($unifiedSearchModuleDisplay as $module => $data) {
+                    if (!empty($data['visible'])) {
+                        $modulesToSearch[$module] = $beanList[$module];
+                    }
+                }
+            }
+            $current_user->setPreference('globalSearch', $modulesToSearch, 'search');
+        }
+
+        return $modulesToSearch;
+    }
+
+    /**
+     *
+     * @param array $modulesToSearch
+     * @param string $searchQuery
+     * @return array
+     * @noinspection DisconnectedForeachInstructionInspection
+     * @noinspection PhpIncludeInspection
+     */
+    private function searchModules(array $modulesToSearch, string $searchQuery): array
+    {
+        global $beanFiles;
+
+        $unifiedSearchModules = $this->getUnifiedSearchModules();
+
+        $moduleResults = [];
+        $moduleCounts = [];
+        $listViewDefs = [];
+
+        if (!empty($searchQuery)) {
+            foreach ($modulesToSearch as $moduleName => $beanName) {
+                require_once $beanFiles[$beanName];
+                $seed = new $beanName();
+
+                $listViewData = new ListViewData();
+
+                // Retrieve the original list view defs and store for processing in case of custom layout changes
+                require(__DIR__ . '/../../../modules/' . $seed->module_dir . '/metadata/listviewdefs.php');
+                $orig_listViewDefs = $listViewDefs;
```

Comment:
Hi @Dillon-Brown probably this was here before, but we could we rename `$orig_listViewDefs` to keep with the standard, something like: `$originalListViewDefs` or similar?

Reply:
Pushed

Reply:
thank you